ui/src/app/subnets/views/FabricDetails/FabricDetailsHeader/FabricDetailsHeader.tsx

```diff
@@ -1,76 +1,33 @@
 import { useState } from "react";
 
-import {
-  Button,
-  Col,
-  ActionButton,
-  Strip,
-  Row,
-} from "@canonical/react-components";
-import { useSelector, useDispatch } from "react-redux";
-import { useHistory } from "react-router-dom";
+import { Button } from "@canonical/react-components";
+
+import FabricDeleteForm from "./FabricDeleteForm";
 
 import SectionHeader from "app/base/components/SectionHeader";
-import { useCycled } from "app/base/hooks";
-import { actions as fabricActions } from "app/store/fabric";
-import fabricSelectors from "app/store/fabric/selectors";
 import type { Fabric } from "app/store/fabric/types";
-import urls from "app/subnets/urls";
 
 type Props = {
   fabric: Fabric;
 };
 
 const FabricDetailsHeader = ({ fabric }: Props): JSX.Element => {
-  const [showConfirm, setShowConfirm] = useState(false);
-  const dispatch = useDispatch();
-  const history = useHistory();
-  const saving = useSelector(fabricSelectors.saving);
-  const saved = useSelector(fabricSelectors.saved);
-
-  const deleteFabric = () => {
-    dispatch(fabricActions.delete(fabric.id));
-  };
-
-  useCycled(saved, () => {
-    if (saved) {
-      history.replace(urls.indexBy({ by: "fabric" }));
-    }
-  });
+  const [showDeleteForm, setShowDeleteForm] = useState(false);
 
   return (
     <SectionHeader
       title={fabric.name}
       buttons={[
-        <Button
-          appearance="neutral"
-          disabled={showConfirm}
-          onClick={() => setShowConfirm(true)}
-        >
+        <Button appearance="neutral" onClick={() => setShowDeleteForm(true)}>
```

Comment:
This should only appear if it's not the default fabric and if you're an admin: https://github.com/canonical-web-and-design/maas-ui/blob/master/legacy/src/app/partials/fabric-details.html#L24

Reply:
I wonder if we should only hide the button for the non-admin case, but include it with an error message for the default fabric case.

Reply:
Yeah that sounds helpful. We should do that for the delete VLAN button as well.